src/main/java/com/wb/exception/CustomExceptionHandler.java

```diff
@@ -0,0 +1,47 @@
+package com.wb.exception;
+
+import java.net.ConnectException;
+import java.net.NoRouteToHostException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import com.wb.payload.ErrorResponse;
+
+@ControllerAdvice
+public class CustomExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ErrorResponse handleIllegalArgumentException(Exception ex, WebRequest request) {
+        List<String> errors = new ArrayList<>();
```

Comment:
`Collections.singletonList()` 
jednolinijkowy sposób na listę jednoelementową.
https://www.geeksforgeeks.org/collections-singletonlist-method-in-java-with-examples/
widze że powtarza się to w kolejnych metodach więc mozesz zrobic osobną metode:
`errorListFromException(Exception e)` a w niej ten singleton list i ją wywołać w miejscu gdzie podajesz `errors`